src/vm/exceptionhandling.cpp

```diff
@@ -4667,59 +4595,38 @@ VOID DECLSPEC_NORETURN UnwindManagedExceptionPass1(PAL_SEHException& ex)
-            firstPassFlags.SetUnwindHasStarted();
-
-            // Tell the tracker that we are starting interleaved handling of the exception.
-            // The interleaved handling is used when an exception unwinding passes through
-            // interleaved managed and native stack frames. In that case, instead of
-            // performing first pass of the unwinding over all the stack range and then 
-            // second pass over the same range, we unwind each managed / native subrange
-            // separately, performing both passes on a subrange before moving to the next one.
-            firstPassFlags.SetIsInterleavedHandling();
+            BOOL success = PAL_VirtualUnwind(&frameContext, NULL);
+            if (!success)
+            {
+                _ASSERTE(!"Thread::VirtualUnwindToFirstManagedCallFrame: PAL_VirtualUnwind failed");
+                EEPOLICY_HANDLE_FATAL_ERROR(COR_E_EXECUTIONENGINE);
+            }
 
-            *currentFlags = firstPassFlags;
+            controlPc = GetIP(&frameContext);
 
+            if (controlPc == 0)
             {
-                GCX_COOP();
-                // Pop all frames that are below the block of native frames and that would be
-                // in the unwound part of the stack when UnwindManagedExceptionPass1 is resumed 
-                // at the next managed frame.
-                UnwindFrameChain(GetThread(), (VOID*)GetSP(&frameContext));
+                // This displays the managed stack in case the unwind has walked out of the stack and
+                // a managed exception was being unwound.
+                DefaultCatchHandler(NULL /*pExceptionInfo*/, NULL /*Throwable*/, TRUE /*useLastThrownObject*/,
+                                    TRUE /*isTerminating*/, FALSE /*isThreadBaseFIlter*/, FALSE /*sendAppDomainEvents*/);
 
-                // We are going to reclaim the stack range that was scanned by the exception tracker
-                // until now. We need to reset the explicit frames range so that if GC fires before
-                // we recreate the tracker at the first managed frame after unwinding the native 
-                // frames, it doesn't attempt to scan the reclaimed stack range.
-                ExceptionTracker* pTracker = GetThread()->GetExceptionState()->GetCurrentExceptionTracker();
-                pTracker->ResetUnwoundExplicitFramesRange();
+                EEPOLICY_HANDLE_FATAL_ERROR(COR_E_EXECUTIONENGINE);                    
             }
 
-            // Now we need to unwind the native frames until we reach managed frames again or the exception is
-            // handled in the native code.
-            StartUnwindingNativeFrames(&frameContext, &ex);
-            UNREACHABLE();
+            UINT_PTR parentSp = GetSP(&frameContext);
+
+            NativeExceptionHolderBase* holder = NativeExceptionHolderBase::FindHolder((void*)sp, (void*)parentSp);
```

Comment:
I think that it would be around line 4613, right before the DefaultCatchHandler. Actually, if the InternalUnhandledExceptionFilter prints the exception stack, the DefaultCatchHandler call can go away after adding the InternalUnhandledExceptionFilter.